### PR TITLE
Refactor RLStatement and TCLStatement to different dialect Statement

### PR DESCRIPTION
**Pull request by strongduanmu**
Ref #7170.

Changes proposed in this pull request:
- Refactor RLStatement and TCLStatement to different dialect Statement

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/sqlserver/tcl/SQLServerSetAutoCommitStatement.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import org.apache.shardingsphere.sql.parser.sql.common.statement.tcl.SetAutoCommitStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.tcl.TCLStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.sqlserver.SQLServerStatement;
+
+/**
+ * SQLServer set auto commit statement.
+ */
+@Getter
+@Setter
+public final class SQLServerSetAutoCommitStatement extends SetAutoCommitStatement implements TCLStatement, SQLServerStatement {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/sqlserver/tcl/SQLServerSetAutoCommitStatement.java:31]: Do you think it is better to move `TCLStatement` to its father class `SetAutoCommitStatement` if they extend the same abstract class?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/sqlserver/tcl/SQLServerSetAutoCommitStatement.java (hunk 1 above)
Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/sqlserver/tcl/SQLServerSetAutoCommitStatement.java:31]: > Do you think it is better to move `TCLStatement` to its father class `SetAutoCommitStatement` if they extend the same abstract class?

@tristaZero Yes, this is a mistake in my refactor, and I will optimize it.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/tcl/MySQLBeginTransactionStatement.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+package org.apache.shardingsphere.sql.parser.sql.dialect.statement.mysql.tcl;
+
+import org.apache.shardingsphere.sql.parser.sql.common.statement.tcl.BeginTransactionStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.tcl.TCLStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.mysql.MySQLStatement;
+
+/**
+ * MySQL begin transaction statement.
+ */
+public final class MySQLBeginTransactionStatement extends BeginTransactionStatement implements TCLStatement, MySQLStatement {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/tcl/MySQLBeginTransactionStatement.java:27]: BeginTransactionStatement implements TCLStatement ?

Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/tcl/MySQLBeginTransactionStatement.java:27]: Move `implements TCLStatement` to `BeginTransactionStatement` ?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/tcl/OracleCommitStatement.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+package org.apache.shardingsphere.sql.parser.sql.dialect.statement.oracle.tcl;
+
+import org.apache.shardingsphere.sql.parser.sql.common.statement.tcl.CommitStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.tcl.TCLStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.oracle.OracleStatement;
+
+/**
+ * Oracle commit statement.
+ */
+public final class OracleCommitStatement extends CommitStatement implements TCLStatement, OracleStatement {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/tcl/OracleCommitStatement.java:27]: CommitStatement implements TCLStatement?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/tcl/MySQLBeginTransactionStatement.java (hunk 2 above)
Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/tcl/MySQLBeginTransactionStatement.java:27]: > Move `implements TCLStatement` to `BeginTransactionStatement` ?

@tristaZero This sounds great, and extracting to the parent class will make the child class more concise. 👍

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/tcl/OracleCommitStatement.java (hunk 3 above)
Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/tcl/OracleCommitStatement.java:27]: > CommitStatement implements TCLStatement?

@tristaZero I will optimize it.

**Review by tristaZero — APPROVED** [review]
Nice job.

**Pull request merged**